misc/scripts/build.sh

```diff
@@ -56,6 +56,38 @@ function clean_builddir() {
     sudo rm -f "${STAGEDIR:?}/${pacname}.deb"
 }
 
+function check_gen_dep() {
+    { ignore_stack=false; set -o pipefail; trap stacktrace ERR RETURN; }
+    local onlyname="${1}" onlyarch="${2}" onlyreal="${3}" onlywhere="${4}"
+    if [[ ${onlyname} == *":${onlyarch}" ]]; then
+        if [[ -z "$(aptitude search --quiet --disable-columns "?exact-name(${onlyname%:*})?architecture(${onlyarch})" -F "%p")" ]]; then
+            if [[ -z "$(aptitude search --quiet --disable-columns "?exact-name(${onlyname%:*})?architecture(all)" -F "%p")" ]]; then
+                if [[ -z "$(aptitude search --quiet --disable-columns "?provides(^${onlyname%:*}$)?architecture(${onlyarch})" -F "%p")" ]]; then
+                    if [[ -z "$(aptitude search --quiet --disable-columns "?provides(^${onlyname%:*}$)?architecture(all)" -F "%p")" ]]; then
+                        fancy_message sub $"%b [required]" "${CYAN}${onlyreal}${NC} ${RED}✗${NC}"
+                        echo "${onlyreal}" >> "${onlywhere}"
+                        return 0
+                    fi
+                fi
+            fi
+        fi
+    else
+        if [[ -z "$(apt-cache search --no-generate --names-only "^${onlyname}\$" 2> /dev/null || apt-cache search --names-only "^${onlyname}\$")" ]]; then
+            if [[ -z "$(aptitude search --quiet --disable-columns "?exact-name(${onlyname})?architecture(${onlyarch})" -F "%p")" ]]; then
+                if [[ -z "$(aptitude search --quiet --disable-columns "?exact-name(${onlyname})?architecture(all)" -F "%p")" ]]; then
+                    if [[ -z "$(aptitude search --quiet --disable-columns "?provides(^${onlyname}$)?architecture(${onlyarch})" -F "%p")" ]]; then
+                        if [[ -z "$(aptitude search --quiet --disable-columns "?provides(^${onlyname}$)?architecture(all)" -F "%p")" ]]; then
+                            fancy_message sub $"%b [required]" "${CYAN}${onlyreal}${NC} ${RED}✗${NC}"
+                            echo "${onlyreal}" >> "${onlywhere}"
+                            return 0
+                        fi
+                    fi
+                fi
+            fi
+        fi
+    fi
```

Comment:
```suggestion
    if [[ ${onlyname} == *":${onlyarch}" &&
        -z "$(aptitude search --quiet --disable-columns "?exact-name(${onlyname%:*})?architecture(${onlyarch})" -F "%p")" &&
        -z "$(aptitude search --quiet --disable-columns "?provides(^${onlyname%:*}$)?architecture(${onlyarch})" -F "%p")" ]] \
        || [[ -z "$(apt-cache search --no-generate --names-only "^${onlyname}\$" 2> /dev/null || apt-cache search --names-only "^${onlyname}\$")" &&
        -z "$(aptitude search --quiet --disable-columns "?exact-name(${onlyname})?architecture(${onlyarch})" -F "%p")" &&
        -z "$(aptitude search --quiet --disable-columns "?provides(^${onlyname}$)?architecture(${onlyarch})" -F "%p")" ]]; then
        fancy_message sub $"%b [required]" "${CYAN}${onlyreal}${NC} ${RED}✗${NC}"
        echo "${onlyreal}" >> "${onlywhere}"
        return 0
    fi

    return 1
```

Reply:
Not sure why the else was removed, this breaks the logic

Reply:
Is the or-statement not equivalent to the if-else?
Both branches are doing the exact same thing

Reply:
I see it now the suggestion format just was impossible to read

it should not have the return 1 tho

Reply:
Updated